util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/multitenancy/impl/H2LazyMultiTenantConnectionProvider.java

```diff
@@ -61,14 +70,34 @@ public ConnectionProvider selectConnectionProvider(String tenantIdentifier) {
 		return connectionProviderImpl;
 	}
 
+	@SuppressWarnings("deprecation")
 	private Map<String, ConnectionProvider> getOrCreateDelegates() {
 		if ( !delegates.isEmpty() ) {
 			return delegates;
 		}
-		for ( String tenantId : tenantIds ) {
-			ConnectionProvider connectionProvider = ConnectionProviderBuilder.buildConnectionProvider( tenantId );
-			delegates.put( tenantId, connectionProvider );
+
+		Properties properties = Environment.getProperties();
+		HashMap<String, Object> map = new HashMap<>();
+		DatabaseContainer.configuration().add( map );
+		// properties should be a copy, so we edit it as we want!
+		properties.putAll( map );
+
+		try {
+			Method buildConnectionProvider = ConnectionProviderBuilder.class.getDeclaredMethod( "buildConnectionProvider",
+					Properties.class, boolean.class );
+			buildConnectionProvider.setAccessible( true );
+
+			for ( String tenantId : tenantIds ) {
+				properties.put( JdbcSettings.URL, String.format( Locale.ROOT, URL_FORMAT, tenantId ) );
+				ConnectionProvider connectionProvider =
+						(ConnectionProvider) buildConnectionProvider.invoke( null, properties, false );
+				delegates.put( tenantId, connectionProvider );
+			}
+		}
+		catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+			throw new IllegalStateException( "Unable to prepare connection providers", e );
```

Comment:
Note, there's also a PR https://github.com/hibernate/hibernate-orm/pull/7672 so we may not need to do something like that